docs/src/GettingStarted.md

```diff
@@ -0,0 +1,225 @@
+```
+
+#### 2. Choose the Method and Call `find_zero`
+```julia
+# Provide a single initial guess for Newton's method
+method = NewtonsMethod(1.0)
+
+# The function f_with_deriv is passed to find_zero
+sol = find_zero(f_with_deriv, method)
+
+println("Root found: ", sol.root) # Expected: 2.0
+```
+
+### Specific Example: Brent's Method for Robust Root Finding
+
+Brent's method combines the bisection method, secant method, and inverse quadratic interpolation. It provides superlinear convergence while maintaining the robustness of bracketing methods.
+
+#### 1. Define Your Function
+```julia
+# This function finds the root of f(x) = x^3 - 2.
+f(x) = x^3 - 2
+```
+
+#### 2. Choose the Method and Call `find_zero`
+```julia
+# Provide a bracketing interval where f(x0) and f(x1) have opposite signs
+method = BrentsMethod(-1.0, 2.0)  # f(-1) = -3, f(2) = 6
+
+# Solve the root-finding problem
+sol = find_zero(f, method)
+
+println("Root found: ", sol.root) # Expected: ≈ 1.259921
+println("Converged: ", sol.converged)
+```
+
+---
+
+## High-Performance and GPU Computing 🚀
+
+RootSolvers.jl is designed for high-performance computing, supporting broadcasting over custom data structures and GPU acceleration. This makes it ideal for solving many problems in parallel.
+
+### Broadcasting with Abstract Types
+The package works seamlessly with any abstract type that supports broadcasting, making it well-suited for scientific domains like climate modeling.
+
+**Example: Solving over a custom field type**
+```julia
+using RootSolvers
+
+# Example using regular arrays to represent a field grid
+x0 = rand(10, 10)  # A 10x10 field of initial guesses
+x1 = x0 .+ 1.0     # A second field of initial guesses
+
+# Define a function that operates element-wise on the field
+f(x) = x.^2 .- 2.0
+
+# Solve the root-finding problem across the entire field
+method = SecantMethod(x0, x1)
+sol = find_zero.(f, method, CompactSolution()) # sol is an Array of structs
+
+# Results
+# Use getproperty.() to extract the fields from each struct in the array
+converged_field = getproperty.(sol, :converged)
+root_field = getproperty.(sol, :root)
+
+println("All converged: ", all(converged_field))
+println("Root field shape: ", size(root_field))
+```
+
+### GPU Acceleration for Batch Processing
+You can achieve significant speedups by running large batches of problems on a GPU.
+
+**GPU Usage Tips:**
+- **Use `CompactSolution`:** Only `CompactSolution` is GPU-friendly. `VerboseSolution` is for CPU debugging only.
+- **GPU-Compatible Function:** Ensure your function `f(x)` uses only GPU-supported operations.
+- **Minimize Data Transfer:** Keep initial guesses and results on the GPU.
+
+**Example: 1 Million problems on the GPU**
+```julia
+using RootSolvers, CUDA
+
+# Create GPU arrays for batch processing
+x0 = CUDA.fill(1.0f0, 1000, 1000)  # 1M initial guesses on GPU
+x1 = CUDA.fill(2.0f0, 1000, 1000)  # Second initial guesses
+
+# Define GPU-compatible function
+f(x) = x.^3 .- x .- 2.0
```

Comment:
```suggestion
f(x) = x^3 - x - 2
```
Why is broadcasting needed here? Isn't this function called in parallel at every point of the solution array, rather than once for the entire array?

Also, `- 2.0` introduces a `Float32 -> Float64` conversion. Use `- 2` to avoid this.